plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -963,17 +963,29 @@ func (c *PodAffinityChecker) InterPodAffinityMatches(pod *v1.Pod, meta algorithm
 	if node == nil {
 		return false, nil, fmt.Errorf("node not found")
 	}
-	if !c.satisfiesExistingPodsAntiAffinity(pod, meta, nodeInfo) {
-		return false, []algorithm.PredicateFailureReason{ErrPodAffinityNotMatch}, nil
+	if satisfy, error := c.satisfiesExistingPodsAntiAffinity(pod, meta, nodeInfo); !satisfy {
+		switch error {
```

Comment:
If you do what I said regarding returning errors, you won't need this switch. You can always
return false, []algorithm.PredicateFailureReason{ErrPodAffinityNotMatch, ErrExistingPodsAntiAffinityRulesNotMatch}, error